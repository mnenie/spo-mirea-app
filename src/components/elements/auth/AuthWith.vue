<script setup>
import {ref} from 'vue'
import {useAuth} from '@/stores/auth.store'

const auth = useAuth()
const onGoogleSubmit = async () => {
  await auth.userLoginWithGoogle()
}

const authIcons = ref([
  {id: 1, name: 'google', url: '/icons/google.png', submit: onGoogleSubmit}
])
</script>

<template>
  <div class="auth_with">
    <div class="line-section">
      <div class="line"></div>
      <span class="line_name">Зайти с помощью</span>
      <div class="line"></div>
    </div>
    <div class="btns-section">
      <div v-for="i in authIcons" :key="i.id" @click="i.submit" class="btn">
        <img :src="i.url" :alt="i.name">
      </div>
    </div>
  </div>
</template>


<style scoped>
.auth_with{
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 14px;
  margin-bottom: 86px;
}
.line-section{
  display: flex;
  align-items: center;
  gap: 11px;
  width: 100%;
}
.line{
  width: 100%;
  height: 0.5px;
  background: var(--black-opacity-color);
}
.line_name{
  white-space: nowrap;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  color: var(--text-color);
}
.btns-section{
  display: flex;
  align-items: center;
  gap: 20px;
  justify-content: center;
}
.btn{
  width: 46px;
  height: 46px;
  border-radius: 15px;
  border: 1px solid var(--border-color);
  display: flex;
  align-items: center;
  justify-content: center;
}
.btn > img{
  width: 26px;
  height: 26px;
}
</style>