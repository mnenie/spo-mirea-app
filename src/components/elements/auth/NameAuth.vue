<script setup>

</script>

<template>
  <h1 class="name">
    Знакомства +
  </h1>
</template>

<style scoped>
.name{
  font-size: 30px;
  color: var(--red-color);
  letter-spacing: 150%;
  font-weight: 400;
  margin-bottom: 124px;
}
</style>