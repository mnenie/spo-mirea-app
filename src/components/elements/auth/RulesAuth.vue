<script setup>
import helperRedirectTinder from '@/helpers/helperRedirectTinder'
const { redirect } = helperRedirectTinder()
</script>

<template>
  <div class="rules">
    <span @click="redirect" class="span">
      Правила пользования
    </span>
    <span @click="redirect" class="span">
      Политика конфиденциальности
    </span>
  </div>
</template>

<style scoped>
.rules {
  display: flex;
  flex-direction: column;
  gap: 13px;
  align-items: center;
}
.span{
  color: var(--red-color);
  font-weight: 400;
  font-size: 14px;
}
.span:hover{
  text-decoration: underline var(--red-color);
}
</style>