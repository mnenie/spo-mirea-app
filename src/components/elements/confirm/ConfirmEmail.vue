<script setup>
import { useAuth } from '@/stores/auth.store'
import { useRoute } from 'vue-router';
import { useField, useForm } from 'vee-validate';
import { toTypedSchema } from '@vee-validate/zod';
import * as zod from 'zod';

const route = useRoute()
const auth = useAuth()

const validationSchema = toTypedSchema(
  zod.object({
    email: zod.string({required_error: "*Поле email обязательно для заполнения"}).nonempty('*Поле email не должно быть пустым').email({ message: '*Некорректный адрес email' }),
    password: zod.string({required_error: "*Поле password обязательно для заполнения"}).nonempty('*Поле password не должно быть пустым').min(8, { message: '*Пароль должен быть не менее 8 символов' }),
  })
);

const { handleSubmit, errors } = useForm({
  validationSchema,
});

const { value: email } = useField('email');
const { value: password } = useField('password');

const onSubmit = handleSubmit(async (values) => {
  const userData = {
    email: email.value,
    password: password.value
  }
  if (route.query.from === 'registration') {
    await auth.userRegistrationWithEmail(userData)
  }
  else if (route.query.from === 'auth') {
    await auth.userLoginWithEmail(userData)
  }
})
</script>

<template>
  <div class="confirm">
    <div class="text">
      <h1>Мой Аккаунт</h1>
      <p>Пожалуйста, введите свой действующий email и введите пароль.</p>
      <div style="display: flex; flex-direction: column; margin-bottom: 15px; gap: 5px;">
        <DefaultInput v-model="email" :placeholder="'Введите email'"/>
        <span v-if="errors.email" style='color: var(--red-color); font-size: 10px;'>{{ errors.email }}</span>
      </div>
      <div style="display: flex; flex-direction: column; margin-bottom: 15px; gap: 5px;">
        <DefaultInput v-model="password" type="password" :placeholder="'Введите пароль'"/>
        <span v-if="errors.password" style='color: var(--red-color); font-size: 10px;'>{{ errors.password }}</span>
      </div>
    </div>
    <RedButton  @click="onSubmit">Продолжить</RedButton>
  </div>
</template>


<style scoped>
.confirm {
  display: flex;
  flex-direction: column;
  gap: 40px;
}

.text {
  display: flex;
  flex-direction: column;
}

.text>h1 {
  font-weight: 400;
  font-size: 30px;
  color: var(--text-color);
  margin-bottom: 6px;
}

.text>p {
  font-weight: 400;
  font-size: 12px;
  color: var(--text-color);
  margin-bottom: 20px;
}
</style>