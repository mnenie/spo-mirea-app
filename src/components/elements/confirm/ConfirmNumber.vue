<script setup>
import { ref, onMounted } from 'vue'
const number = ref('')
import { auth } from '@/firebase/index'
import { RecaptchaVerifier, signInWithPhoneNumber } from 'firebase/auth';
import { CODE_ROUTE } from '@/utils/consts';
import { useRouter } from 'vue-router';

const router = useRouter()

const recaptchaVerifier = ref(null);
const recaptchaWidgetId = ref(null);
const confirmResult = ref(null);

onMounted(() => {
  auth.useDeviceLanguage();
  recaptchaVerifier.value = new RecaptchaVerifier(auth, "recaptcha-container", {
    'size': 'invisible',
  });
  recaptchaVerifier.value.render().then((widgetId) => {
    recaptchaWidgetId.value = widgetId;
  });
});

const onSubmit = async() => {
  const appVerifier = recaptchaVerifier.value;
  try{
    const confirmationResult = await signInWithPhoneNumber(auth, number.value, appVerifier);
    confirmResult.value = confirmationResult;
    if(confirmResult.value){
      router.push(CODE_ROUTE)
    }
  } catch (error) {
    console.error("Error sending verification code", error);
  }
}


</script>

<template>
  <div id="recaptcha-container"></div>
  <div class="confirm">
    <div class="text">
      <h1>Мой номер</h1>
      <p>Пожалуйста, введите свой действующий номер телефона. Мы вышлем вам 4-значный код для подтверждения вашей
        учетной записи.</p>
      <DefaultInput v-model="number" :placeholder="'+7 (XXX) XXX-XX-XX'" />
    </div>  
    <RedButton @click="onSubmit" id="sign-in-button">Продолжить</RedButton>
  </div>
</template>


<style scoped>
.confirm {
  display: flex;
  flex-direction: column;
  gap: 54px;
}

.text {
  display: flex;
  flex-direction: column;
}

.text>h1 {
  font-weight: 400;
  font-size: 30px;
  color: var(--text-color);
  margin-bottom: 6px;
}

.text>p {
  font-weight: 400;
  font-size: 12px;
  color: var(--text-color);
  margin-bottom: 20px;
}
</style>