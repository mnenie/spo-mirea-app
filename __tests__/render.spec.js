import {describe, it, expect, beforeAll} from 'vitest'
import {mount, shallowMount} from "@vue/test-utils"
import { createPinia } from 'pinia'

import NameAuth from '@/components/elements/auth/NameAuth.vue'
import RulesAuth from '@/components/elements/auth/RulesAuth.vue'
import AuthWith from '@/components/elements/auth/AuthWith.vue'
import ConfirmEmail from '@/components/elements/confirm/ConfirmEmail.vue'
import ConfirmNumber from '@/components/elements/confirm/ConfirmNumber.vue'

//config
let wrapper;
beforeAll(() => {
  wrapper = shallowMount(AuthWith, {
    global: {
      plugins: [createPinia()],
    },
  });
})

describe('NameAuth', () => {
  it('should render correctly', () => {
    wrapper = mount(NameAuth)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('RulesAuth', () => {
  it('should render correctly', () => {
    wrapper = mount(RulesAuth)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('AuthWith', () => {
  it('should render correctly', () => {
    wrapper = mount(AuthWith)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('ConfirmEmail', () => {
  it('should render correctly', () => {
    wrapper = mount(ConfirmEmail)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('ConfirmNumber', () => {
  it('should render correctly', () => {
    wrapper = mount(ConfirmNumber)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

