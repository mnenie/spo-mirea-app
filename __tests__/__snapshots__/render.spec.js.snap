// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthWith > should render correctly 1`] = `
"<div data-v-cc9dd5da="" class="auth_with">
  <div data-v-cc9dd5da="" class="line-section">
    <div data-v-cc9dd5da="" class="line"></div><span data-v-cc9dd5da="" class="line_name">Зайти с помощью</span>
    <div data-v-cc9dd5da="" class="line"></div>
  </div>
  <div data-v-cc9dd5da="" class="btns-section">
    <div data-v-cc9dd5da="" class="btn"><img data-v-cc9dd5da="" src="/icons/google.png" alt="google"></div>
  </div>
</div>"
`;

exports[`ConfirmEmail > should render correctly 1`] = `
"<div data-v-a90709b2="" class="confirm">
  <div data-v-a90709b2="" class="text">
    <h1 data-v-a90709b2="">Мой Аккаунт</h1>
    <p data-v-a90709b2="">Пожалуйста, введите свой действующий email и введите пароль.</p>
    <div data-v-a90709b2="" style="display: flex; flex-direction: column; margin-bottom: 15px; gap: 5px;">
      <defaultinput data-v-a90709b2="" placeholder="Введите email"></defaultinput>
      <!--v-if-->
    </div>
    <div data-v-a90709b2="" style="display: flex; flex-direction: column; margin-bottom: 15px; gap: 5px;">
      <defaultinput data-v-a90709b2="" type="password" placeholder="Введите пароль"></defaultinput>
      <!--v-if-->
    </div>
  </div>
  <redbutton data-v-a90709b2="">Продолжить</redbutton>
</div>"
`;

exports[`ConfirmNumber > should render correctly 1`] = `
"<div data-v-cc0f9ae9="" class="confirm">
  <div data-v-cc0f9ae9="" class="text">
    <h1 data-v-cc0f9ae9="">Мой номер</h1>
    <p data-v-cc0f9ae9="">Пожалуйста, введите свой действующий номер телефона. Мы вышлем вам 4-значный код для подтверждения вашей учетной записи.</p>
    <defaultinput data-v-cc0f9ae9="" modelvalue="" placeholder="+7 (XXX) XXX-XX-XX"></defaultinput>
  </div>
  <redbutton data-v-cc0f9ae9="">Продолжить</redbutton>
</div>"
`;

exports[`NameAuth > should render correctly 1`] = `"<h1 data-v-7d796deb="" class="name"> Знакомства + </h1>"`;

exports[`RulesAuth > should render correctly 1`] = `"<div data-v-1f0eb9a3="" class="rules"><span data-v-1f0eb9a3="" class="span"> Правила пользования </span><span data-v-1f0eb9a3="" datatype="redirect" class="span"> Политика конфиденциальности </span></div>"`;
